zeus/device/soc/jetson.py

```diff
@@ -0,0 +1,280 @@
+    """Reads power directly from a sysfs path."""
+
+    def __init__(self, power_path: Path) -> None:
+        """Initialize DirectPower paths."""
+        self.power_path: Path = power_path
+
+    def measure_power(self) -> float:
+        """Measure power by reading from sysfs paths.
+
+        Units: mW.
+        """
+        power: float = float(self.power_path.read_text().strip())
+        return power
+
+
+class VoltageCurrentProduct(PowerMeasurementStrategy):
+    """Computes power as product of voltage and current, read from two sysfs paths."""
+
+    def __init__(self, voltage_path: Path, current_path: Path) -> None:
+        """Initialize VoltageCurrentProduct paths."""
+        self.voltage_path: Path = voltage_path
+        self.current_path: Path = current_path
+
+    def measure_power(self) -> float:
+        """Measure power by reading from sysfs paths.
+
+        Units: mW.
+        """
+        voltage: float = float(self.voltage_path.read_text().strip())
+        current: float = float(self.current_path.read_text().strip())
+        return voltage * current
+
+
+@dataclass
+class JetsonMeasurement(soc_common.SoCMeasurement):
+    """Represents energy measurements for Jetson Nano subsystems."""
+
+    cpu_energy_mj: float = 0.0
+    gpu_energy_mj: float = 0.0
+    total_energy_mj: float = 0.0
+
+    def __sub__(self, other: JetsonMeasurement) -> JetsonMeasurement:
+        """Return a new JetsonMeasurement with subtracted field values."""
+        if not isinstance(other, JetsonMeasurement):
+            raise ValueError("Other must be a JetsonMeasurement object.")
+
+        return JetsonMeasurement(
+            cpu_energy_mj=self.cpu_energy_mj - other.cpu_energy_mj,
+            gpu_energy_mj=self.gpu_energy_mj - other.gpu_energy_mj,
+            total_energy_mj=self.total_energy_mj - other.total_energy_mj,
+        )
+
+    def zeroAllFields(self) -> None:
+        """Set all internal measurement values to zero."""
+        self.cpu_energy_mj = 0.0
+        self.gpu_energy_mj = 0.0
+        self.total_energy_mj = 0.0
+
+
+class Jetson(soc_common.SoC):
+    """An interface for obtaining the energy metrics of a Jetson Nano processor."""
+
+    def __init__(self) -> None:
+        """Initialize Jetson monitoring object."""
+        super().__init__()
+
+        # Maps each power rail ('cpu', 'gpu', and 'total') to a power measurement strategy
+        self.power_measurement: dict[
+            str, PowerMeasurementStrategy
+        ] = self._discover_available_metrics()
+
+        # Spawn polling process
+        context = mp.get_context("spawn")
+        self.command_queue = context.Queue()
+        self.result_queue = context.Queue()
+        self.process = context.Process(
+            target=_polling_process_async_wrapper,
+            args=(self.command_queue, self.result_queue, self.power_measurement),
+        )
+        self.process.start()
+
+        atexit.register(self._stop_process)
```

Comment:
I noticed that all the CI tests are failing with a hang. I think it's because when `Jetson` is instantiated inside `get_soc()`, it'll spawn the polling thread even if there's no Jetson chip around.

`Jetson.__init__` should raise a `ZeusJetsonInitError` if it realizes that there's no Jetson device, where `ZeusJetsonInitError` is defined inside `jetson.py` like this:

```python
class ZeusJetsonInitError(ZeusSoCInitError):
    """Import error for Jetson initialization failures."""

    def __init__(self, message: str) -> None:
        """Initialize Zeus Exception."""
        super().__init__(message)
```